cm_tools.module

```diff
@@ -64,6 +64,72 @@ function cm_tools_permission() {
   );
 }
 
+/**
+ * Implements hook_menu().
+ */
+function cm_tools_menu() {
+  $items['cm_tools/monitoring/%'] = array(
+    'page callback' => 'cm_tools_monitoring_page_callback',
+    'access callback' => 'cm_tools_monitoring_access_callback',
+    'access arguments' => array(2),
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_init().
+ */
+function cm_tools_init() {
+  // Our monitoring page is never in maintenance mode.
+  if ($_GET['q'] === 'cm_tools/monitoring/' . cm_tools_get_uptime_path_token()) {
```

Comment:
Would this be better than comparing the URL, maybe?

```suggestion
  $item = menu_get_item();
  if ($item && $item['page_callback`] === 'cm_tools_monitoring_page_callback') {
```

Reply:
I'd argue for the simple string comparison. I feel like the menu system is full of opportunities for people to do weird things (`menu_get_item` has a static cache of results for example.

The string comparison is clearer and less likely to have weird side effects imho.

Reply:
Isn't that the point then though? You want to target the monitoring page, not the path that we currently believe it sits at. If someone alters the menu or menu item, you only want to hook into this particular route, regardless of where it sits.

Reply:
I want this hook init implementation to be as simple as possible, ideally without reaching out to too many Drupal APIs or causing too many side-effects. It's likely to be running in maintenance mode because the database is not in a consistent state etc.

If someone is doing a menu alter to move the path this module provides, then they can darn well hook_module_implements alter this init hook away and replicate the logic in theirs.

Reply:
All good justification ☺️